### DISCUSSION TITLE
How do I get webpack to include modules under /packages in the bundle?

### DISCUSSION THREAD

**Discussion by sebastiangug** [vercel/turborepo#1667] [Help] [answered]
I'm elaborating the issue I'm having in this SO thread: https://stackoverflow.com/questions/73294477/

But to sum it up, when having a non-next project, like a nestjs project and trying to build up one of the applications for development serving or for production, it doesn't include/work with the packages under /packages. 

Building all the packages as part of a separate process or a parallel process isn't a viable development experience, the equivalent would be if you make a change to the UI package and you have to cancel your dev. server and then rebuild the UI package and then serve the app again. 

Is there a way or an example of a nestjs app using a package? 

It works with VITE + SWC but then vite rebuilds the application for every request causing schema duplicates in GQL and crashing the server.

Repository where this can be reproduced/played with: 

https://github.com/sebastiangug/turborepo-nestjs

**Comment by nathanhammond** [comment, marked as answer]
As a thought experiment, how would you imagine doing this in the situation where this was a stand-alone `npm` module you needed to update? Exactly as you described: pulling down the dev server, running the install, and then standing the server back up.

The way node modules and workspaces interact is exactly the same.

Now, it's _almost always_ possible to subvert the boundaries that are _intended_ to create isolation at module boundaries, but that will require a seemingly magical incantation particular to every project and tech stack—and is entirely separate from Turborepo.

This question is far-better-suited for a nestjs/Webpack forum.

**Reply by nathanhammond** [reply]
For *just* build stuff (not dev server) you can make sure that you set `^build` in `turbo.json` to get topological dependencies to build first, so their assets are available and transpiled. (export token)

**Reply by sebastiangug** [reply]
@nathanhammond  but isn't this exactly the way the starter template of turborepo works? it uses /packages/ui more like shared source code rather than shared modules?

That's the pattern I'm after and I'm intentionally avoiding converting them into fully standalone compiled modules. 

With Vite & SWC this works out of the box, but I had to give that up due to an issue with the nestjs implementation in particular.

From the comments I've seen, that next starter is using webpack to achieve that under the hood -- I'm just not sure how, the actual configuration as turborepo is very next-centric.

**Reply by nathanhammond** [reply]
Our goal is, in short, not to get in the way. I don't believe that we are _preventing_ this from working.

Deriving the magic incantation for every single permutation of "how to make X framework play nicely with workspaces" will remain beyond the scope of Turborepo (and would be a nestjs/webpack question in your case), but, if you discover the incantation and want to share it, we would consider a PR for a starter repo which includes that incantation!

**Reply by nathanhammond** [reply]
Also, Turborepo has no framework specific logic embedded. We're not enabling Next.js in particular (other than by having a starter template which we provide for the most-used frameworks). The work to make workspaces play nicely with dev builds is entirely done by the Next.js team to support the (pnpm/npm/yarn) feature and implementations.
